app/src/main/kotlin/compiler/intermediate/generators/FunctionDetailsGenerator.kt

```diff
@@ -10,12 +10,18 @@ interface FunctionDetailsGenerator : VariableAccessGenerator {
         val result: IFTNode?
```

Comment:
The calls which resume a generator function will almost certainly return two values (the state and the passed value), so we may as well already add `val secondResult: IFTNode? = null` here.

Reply:
True, but I'm not sure if `IFTNode` would be sufficient or should we use a `ControlFlowGraph` for storing the state - it should restore potentially large memory state on heap and set `RIP` to appropriate address - even if we add some `memcpy` as `IFTNode`, these are at least two separate operations

Reply:
Copying won't be necessary. I posted the details on our group chat.